Mathlib/CategoryTheory/Limits/FunctorCategory.lean

```diff
@@ -445,3 +446,127 @@ def colimitIsoSwapCompColim [HasColimitsOfShape J C] (G : J ⥤ K ⥤ C) :
 #align category_theory.limits.colimit_iso_swap_comp_colim CategoryTheory.Limits.colimitIsoSwapCompColim
 
 end
+
+-- variable {C : Type u} [Category.{v} C] {D : Type u'} [Category.{v'} D]
+
+--def typefunc (C : Type u) [Category.{v} C] : Type (max v (u + 1)) := C ⥤ (Type u)
+
+--instance blah : Quiver (typefunc C) := sorry
+
+namespace FunctorProd
+
+/-- `prod X Y` is the explicit binary product of simplicial sets `X` and `Y`. -/
+def prod (F G : C ⥤ (Type u)) : C ⥤ (Type u) where
+  obj n := F.obj n × G.obj n
+  map f a := (F.map f a.1, G.map f a.2)
+
+  /-- The first projection of `prod X Y`, onto X. -/
+@[simps]
+def prod.fst {X Y : C ⥤ (Type u)} : (prod X Y) ⟶ X where
+  app _ a := a.1
+
+/-- The second projection of `prod X Y`, onto Y. -/
+@[simps]
+def prod.snd {X Y : C ⥤ (Type u)} : (prod X Y) ⟶ Y where
+  app _ a := a.2
+
+/-- The binary fan whose point is `prod X Y`. -/
+@[simps!]
+def binaryProductCone (X Y : C ⥤ (Type u)) : BinaryFan X Y :=
+  BinaryFan.mk (prod.fst) (prod.snd)
+
+/-- The morphism from the point of any binary fan to `prod X Y`. -/
+@[simps]
+def binaryProductLift {X Y : C ⥤ (Type u)} (s : BinaryFan X Y) : s.pt ⟶ prod X Y where
+  app n x := ((s.fst).app n x, (s.snd).app n x)
```

Comment:
For future applications of this, I would suggest defining ```def natTransProd {F F₁ F₂ : C ⥤ Type w} (τ₁ : F ⟶ F₁) (τ₂ : F ⟶ F₂) : F ⟶ functorProd F₁ F₂ := sorry``` instead and invoke this `natTransProd` directly in the definition of `binaryProductLimit`.